3DAmsterdam/Assets/Stadsmodel/Buildings/BuildingTileManager.cs

```diff
@@ -1,372 +1,364 @@
+        float X = float.Parse(gameObjectName.Split('_')[0]);
+        float Y = float.Parse(gameObjectName.Split('_')[1]);
+        
+        GameObject container = new GameObject(gameObjectName);
+        container.transform.parent = transform;
+        container.layer = LayerMask.NameToLayer("Panden");
+
+        //positioning container
+        Vector3RD hoekpunt = new Vector3RD(X, Y, 0);
+        double OriginOffset = 500;
+        Vector3RD origin = new Vector3RD(hoekpunt.x + (OriginOffset), hoekpunt.y + (OriginOffset), 0);
+        Vector3 UnityOrigin = CoordConvert.RDtoUnity(origin);
+        container.transform.localPosition = UnityOrigin;
+        double Rotatie = CoordConvert.RDRotation(origin);
+        container.transform.Rotate(Vector3.up, (float)Rotatie);
+
+        //add mesh
+        MeshFilter mf = container.AddComponent<MeshFilter>();
+        mf.sharedMesh = mesh;
+        //add material
+        MeshRenderer mr = container.AddComponent<MeshRenderer>();
+        mr.material = defaultMaterial;
+
+        return container;
+    }
+
+    private void AddObjectIDs (Mesh mesh, int meshIndex, string[] objectIDs, GameObject gameObject)
+    {
+        if (objectIDs.Length==0)
+        {
+            return;
+        }
+        ObjectMapping objectMap = gameObject.AddComponent<ObjectMapping>();
+        Dictionary<float, string> objectIDList = new Dictionary<float, string>();
 
+        string[] objects = Regex.Split(objectIDs[meshIndex * 2], ",");
+        for (int i = 0; i < objects.Length - 1; i++)
+        {
+            objectIDList.Add((float)i, objects[i]);
+        }
+        objectMap.Objectenlijst = objectIDList;
+        objectMap.DefaultMaterial = defaultMaterial;
+        objectMap.HighlightMaterial = highLightMaterial;
+        objectMap.SetMesh(mesh);
+
+    }
 
+    private string[] ReadObjectIDs(AssetBundle assetBundle)
+    {
+        string[] objectIDs = new string[0];
+        if (assetBundle.LoadAllAssets<TextAsset>().Length > 0)
+        {
+            
+            TextAsset textAsset = assetBundle.LoadAllAssets<TextAsset>()[0];
+            string objectIDList = textAsset.text;
+            objectIDs = Regex.Split(objectIDList, "\n|\r|\r\n");
+        }
+        return objectIDs;
     }
 
+    private Extent ConvertWGSExtentToRDExtent(Extent wGSExtent)
+    {
+        Vector3RD RDmin = CoordConvert.WGS84toRD(wGSExtent.MinX, wGSExtent.MinY);
+        Vector3RD RDmax = CoordConvert.WGS84toRD(wGSExtent.MaxX, wGSExtent.MaxY);
+
+        int X0 = ((int)Math.Floor(RDmin.x / 1000) * 1000);
+        int Y0 = ((int)Math.Floor(RDmin.y / 1000) * 1000);
+        int X1 = ((int)Math.Floor(RDmax.x / 1000) * 1000);
+        int Y1 = ((int)Math.Floor(RDmax.y / 1000) * 1000);
+
+        Extent rDExtent = new Extent(X0, Y0, X1, Y1);
+        return rDExtent;
+    }
+
+    private float GetRequiredLOD(double distanceFromCamera)
+    {
+        float requiredLOD = -1;
+
+        if (distanceFromCamera < maximumDistanceLOD0)
```

Comment:
wanneer je deze omdraai kan je er volgens mij twee else ifs van maken:

```suggestion:0+12
        if (distanceFromCamera < maximumDistanceLOD2)
        {
                requiredLOD = 2;
        }
        else if (distanceFromCamera < maximumDistanceLOD1)
        {
                requiredLOD = 1;
        }
        else if (distanceFromCamera < maximumDistanceLOD0)
        {
                requiredLOD = 0;
        }
```